File: 853-most-profit-assigning-work/most-profit-assigning-work.py
class Solution:
    def maxProfitAssignment(self, diff: List[int], prof: List[int], work: List[int]) -> int:
        
        l=[]
        for i in range(len(diff)):
            l.append((diff[i],-prof[i]))
        l.sort()

        pq = []
        work.sort()
        j = 0
        ans = 0
        for i in work:
            while j<len(l) and l[j][0]<=i:
                heapq.heappush(pq,l[j][1])
                j+=1
            
            if pq :
                
                ans -= pq[0]
        return ans

File: 853-most-profit-assigning-work/most-profit-assigning-work.java
class Solution {
    public int maxProfitAssignment(int[] diff, int[] prof, int[] work) {
        int[][] l = new int[diff.length][2];
        for(int i =0;i<diff.length;i++){
            l[i][0] = diff[i];
            l[i][1] = prof[i];
        }

        Arrays.sort(l,new Comparator<int[]>(){
            @Override
            public int compare(int[] a,int[] b){
                return Integer.compare(a[0],b[0]);
            }
        });

        PriorityQueue<Integer> pq = new PriorityQueue<Integer>((a,b)->{
            return b-a;
        });

        int ans = 0,j=0;
        Arrays.sort(work);
        for(int i = 0;i<work.length;i++){
            while (j<l.length && work[i]>=l[j][0]){
                pq.add(l[j][1]);
                j+=1;
            }
            if(pq.size()>0)ans += pq.peek();
        }
        return ans;



    }
}